crates/cheatcodes/src/fs.rs

```diff
@@ -370,6 +387,39 @@ fn ffi(state: &Cheatcodes, input: &[String]) -> Result<FfiResult> {
     })
 }
 
+fn prompt_input(prompt_text: &String) -> Result<String, dialoguer::Error> {
+    Input::new().allow_empty(true).with_prompt(prompt_text).interact_text()
+}
+
+fn prompt_password(prompt_text: &String) -> Result<String, dialoguer::Error> {
+    Password::new().with_prompt(prompt_text).interact()
+}
+
+fn prompt(
+    state: &Cheatcodes,
+    prompt_text: &str,
+    input: fn(&String) -> Result<String, dialoguer::Error>,
+) -> Result<String> {
+    let text_clone = prompt_text.to_string();
+    let timeout = state.config.prompt_timeout;
+    let (send, recv) = mpsc::channel();
+
+    thread::spawn(move || {
+        send.send(input(&text_clone)).unwrap();
+    });
+
+    match recv.recv_timeout(timeout) {
+        Ok(res) => res.map_err(|_| {
+            println!();
+            "I/O error occured".into()
```

Comment:
I think we should return the error as the return value and also log in in an error! trace

Reply:
I'm not sure I understand - do you mean the `prompt()` function should return `Result<String, dialoguer::Error>`?

Reply:
we want to return `err.to_string()` here as the response value so the users has a chance to look at the error in the revert message if the call failed